python/lsst/meas/base/forcedPhotCcd.py

```diff
@@ -41,84 +41,14 @@
-    being measured using the detection information, a set of reference
-    catalogs, from the set of coadds which intersect with the calexp.  It needs
-    the calexp id (e.g.  visit, raft, sensor), but is also uses the tract to
-    decide what set of coadds to use.  The references from the tract whose
-    patches intersect with the calexp are used.
-    """
-
-    def makeDataRefList(self, namespace):
-        """Make self.refList from self.idList
-        """
-        if self.datasetType is None:
-            raise RuntimeError("Must call setDatasetType first")
-        log = logging.getLogger(__name__).getChild("PerTractCcdDataIdContainer")
-        skymap = None
-        visitTract = collections.defaultdict(set)   # Set of tracts for each visit
-        visitRefs = collections.defaultdict(list)   # List of data references for each visit
-        for dataId in self.idList:
-            if "tract" not in dataId:
-                # Discover which tracts the data overlaps
-                log.info("Reading WCS for components of dataId=%s to determine tracts", dict(dataId))
-                if skymap is None:
-                    skymap = namespace.butler.get(namespace.config.coaddName + "Coadd_skyMap")
-
-                for ref in namespace.butler.subset("calexp", dataId=dataId):
-                    if not ref.datasetExists("calexp"):
-                        continue
-
-                    visit = ref.dataId["visit"]
-                    visitRefs[visit].append(ref)
-
-                    md = ref.get("calexp_md", immediate=True)
-                    wcs = lsst.afw.geom.makeSkyWcs(md)
-                    box = lsst.geom.Box2D(lsst.afw.image.bboxFromMetadata(md))
-                    # Going with just the nearest tract.  Since we're throwing all tracts for the visit
-                    # together, this shouldn't be a problem unless the tracts are much smaller than a CCD.
-                    tract = skymap.findTract(wcs.pixelToSky(box.getCenter()))
-                    if imageOverlapsTract(tract, wcs, box):
-                        visitTract[visit].add(tract.getId())
-            else:
-                self.refList.extend(ref for ref in namespace.butler.subset(self.datasetType, dataId=dataId))
-
-        # Ensure all components of a visit are kept together by putting them all in the same set of tracts
-        for visit, tractSet in visitTract.items():
-            for ref in visitRefs[visit]:
-                for tract in tractSet:
-                    self.refList.append(namespace.butler.dataRef(datasetType=self.datasetType,
-                                                                 dataId=ref.dataId, tract=tract))
-        if visitTract:
-            tractCounter = collections.Counter()
-            for tractSet in visitTract.values():
-                tractCounter.update(tractSet)
-            log.info("Number of visits for each tract: %s", dict(tractCounter))
-
-
```

Comment:
pipe_drivers is a dead parrot and is not used by any lsst_distrib code and is entirely gen2. There aren't any tests in pipe_drivers so nothing is trying to load any of these classes anyhow. We have to keep pipe_drivers around because we used to have some gen3 code in there that is still in the upcoming v24 and so we need to warn people to update their pipeline YAML to switch.